zig/private/zig_c_module.bzl

```
@@ -0,0 +1,111 @@
+"""Implementation of the zig_module rule."""
+
+load("@rules_cc//cc:find_cc_toolchain.bzl", "use_cc_toolchain")
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
+load(
+    "//zig/private/common:bazel_builtin.bzl",
+    BAZEL_BUILTIN_ATTRS = "ATTRS",
+)
+load("//zig/private/common:data.bzl", "zig_collect_data", "zig_create_runfiles")
+load("//zig/private/common:translate_c.bzl", "zig_translate_c")
+load("//zig/private/common:zig_cache.bzl", "zig_cache_output")
+load("//zig/private/common:zig_lib_dir.bzl", "zig_lib_dir")
+
+DOC = """\
+Defines a Zig C module.
+
+A Zig C module is a Zig module whose C headers dependencies have been translated
+to zig using `translate-c`, and which only output defines the module's entry
+point.
+
+This rule performs the `translate-c` step but does not perform compilation of
+Zig by itself.
+Instead, modules are compiled at the use-site.
+Zig performs whole program compilation.
+
+**EXAMPLE**
+
+```bzl
+load("@rules_zig//zig:defs.bzl", "zig_c_module")
+
+zig_c_module(
+    name = "my-module",
+    cdeps = [
+        ":cc-library",  # to support `@import("cc-library")`.
```

```suggestion
        ":cc-library",
```
That part isn't quite correct, right? Users wouldn't be able to `@import("cc-library")`. Instead, they'd have to `@import("my-module").some_cc_library_symbol`.